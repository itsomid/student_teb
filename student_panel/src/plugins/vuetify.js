/**
 * plugins/vuetify.js
 *
 * @see vuetify documentation: https://vuetifyjs.com/en/getting-started/installation/#manual-steps
 */

// Styles
import 'vuetify/styles'
/* To remove color pack uncomment below line and comment import 'vuetify/styles'*/
import '@/assets/styles/main.scss'

// Vuetify
import { createVuetify } from 'vuetify'
import { fa, en } from 'vuetify/locale'
import theme from "@/plugins/vuetify-options/theme/theme";
import icons from "@/plugins/vuetify-options/icons/icons";
import { md3 } from 'vuetify/blueprints'

import { VStepperVertical, VStepperVerticalActions, VStepperVerticalItem, } from 'vuetify/labs/VStepperVertical'
export default createVuetify({
  /** @see Vuetify feature guides
   * @tutorial https://vuetifyjs.com/en/introduction/why-vuetify/#feature-guides */
  blueprint: md3,

  locale: {
    locale: 'fa',
    fallback: 'en',
    messages: { fa, en }
  },
  components: {
    VStepperVertical,
    VStepperVerticalActions,
    VStepperVerticalItem
  },
  theme: theme,
  icons: icons,
  defaults: {
    VBtn: {
      color: '', // Set the default button color to primary
      variant: 'flat',  // Other options: 'tonal', 'outlined', 'text', 'elevated'
    },
  },
})
