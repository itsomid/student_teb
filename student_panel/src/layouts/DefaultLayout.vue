<script setup>
import NavBar from "@/components/layouts/navbar/NavBar.vue";
import {onMounted, ref} from "vue";

import DashboardNotifications from "@/components/dashboard/notifications/DashboardNotifications.vue";
import DashboardNotificationModal from "@/components/dashboard/notifications/DashboardNotificationModal.vue";
import {useDisplay} from "vuetify";
import DashboardClassDrawer from "@/components/dashboard/class/DashboardClassDrawer.vue";

import NavigationDrawerDesktop from "@/components/layouts/navigation-drawer/NavigationDrawerDesktop.vue";
import NavigationDrawerMobile from "@/components/layouts/navigation-drawer/NavigationDrawerMobile.vue";
import BottomNavigationMenu from "@/components/layouts/BottomNavigationMenu.vue";
import {useRoute} from "vue-router";
// import Porsline from "@/components/porsline/Porsline.vue";
import {useCart} from "@/composable/useCart";
import AuthSetGradeModal from "@/components/authentication/AuthSetGradeModal.vue";

const drawer = ref(false);
const { lgAndUp } = useDisplay();
const route = useRoute();

const includedRoutesForBottomMenu = ['dashboard','my-courses','store.main','finance.main'];

const { trackUnpaidCartStatus } = useCart({});

onMounted(()=> trackUnpaidCartStatus())
</script>

<template>
  <v-responsive>
    <v-layout class="bg-background rounded rounded-md ss02" width="100vw" style="min-height: 100vh!important;">
      <!--    <HeaderBanner />-->


      <NavigationDrawerDesktop v-if="$vuetify.display.lgAndUp" />
      <NavigationDrawerMobile v-else v-model="drawer" />

      <div v-if="route.name === 'dashboard'">
        <DashboardClassDrawer />
      </div>

      <DashboardNotifications v-if="lgAndUp"></DashboardNotifications>
      <DashboardNotificationModal v-else/>

      <NavBar v-model="drawer"/>
      <v-main class="pb-16 mb-16">
        <v-container  fluid style="max-width: 1600px">
          <router-view v-slot="{ Component, route }">
            <v-fade-transition hide-on-leave mode="out-in">
              <component :is="Component" :key="route.path"  />
            </v-fade-transition>
          </router-view>
        </v-container>
        <BottomNavigationMenu v-if="$vuetify.display.mdAndDown && includedRoutesForBottomMenu.includes(route.name)"/>
<!--        <Porsline v-if="route.name === 'my-courses'"/>-->
        <AuthSetGradeModal />
      </v-main>
    </v-layout>
  </v-responsive>
</template>

<style>

</style>
