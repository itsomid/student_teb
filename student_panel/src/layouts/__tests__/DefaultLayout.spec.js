import {describe , it,expect } from "vitest";
import { mount} from "@vue/test-utils";

import DefaultLayout from "../DefaultLayout.vue"
import {createVuetify} from "vuetify";
const vuetify= createVuetify()

import Drawer from "@/components/layouts/Drawer.vue";
import AppBar from "@/components/layouts/AppBar.vue";

import ResizeObserver from 'resize-observer-polyfill'
window.ResizeObserver = ResizeObserver;

const $route = {
    name: "mock_route",
};
const wrapper  = mount(DefaultLayout , {
    global: {
        plugins: [vuetify],
        stubs:{
            'router-view' : true,
            Drawer : true,
            AppBar : true
        },
        mocks: {
            $route,
        },
    },
})
describe ("DefaultLayout.vue | @/src/layouts" , ()=>{

    it("render DefaultLayout component" , ()=>{
        expect(wrapper.find(".v-application").exists()).toBeTruthy();
    })

    it("render Drawer component in DefaultLayout" , ()=>{
        expect(wrapper.findComponent(Drawer).exists()).toBeTruthy();
    })

    it("render AppBar component in DefaultLayout" , ()=>{
        expect(wrapper.findComponent(AppBar).exists()).toBeTruthy()
    })
})
