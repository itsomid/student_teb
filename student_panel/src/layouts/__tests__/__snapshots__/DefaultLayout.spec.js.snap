// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DefaultLayout.vue | @/src/layouts > renders match to snapshot 1`] = `"<v-app-stub overlaps=\\"\\" fullheight=\\"true\\"></v-app-stub>"`;
